stock_warehouse_flow/models/stock_move.py

```diff
@@ -36,6 +37,14 @@ def _action_confirm(self, merge=True, merge_into=False):
                 move, assign_picking=False
             ).ids
         moves_to_confirm = self.browse(move_ids_to_confirm)
-        return super(StockMove, moves_to_confirm)._action_confirm(
+        res = super(StockMove, moves_to_confirm)._action_confirm(
             merge=merge, merge_into=merge_into
         )
+        # In case the move was already assigned to a picking and action_confirm
+        # was called again, with the flow it may have changed the conditions
+        # and assigned to a different picking. If the initial picking is empty,
+        # marked it as canceled.
+        for old_picking in old_pickings:
+            if not old_picking.move_lines:
+                old_picking.state = "cancel"
```

Comment:
Should we not call `action_cancel` instead?

Odoo is also setting the `is_locked` field:
```python
    def action_cancel(self):
        self.mapped('move_lines')._action_cancel()
        self.write({'is_locked': True})
        self.filtered(lambda x: not x.move_lines).state = 'cancel'
        return True
```